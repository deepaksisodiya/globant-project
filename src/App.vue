<template>
  <div>
    <div class="header">{{ headertext }}</div>
    <button @click="toggleScreen" class="button">{{ buttonText }}</button>
    <ListMovie :movies="movies" v-if="!onMovieListPage" />
    <AddMovie :addNewMovie="addNewMovie" v-if="onMovieListPage" />
  </div>
</template>

<script>
import AddMovie from "./components/AddMovie.vue";
import ListMovie from "./components/ListMovie.vue";

export default {
  name: "app",
  components: {
    AddMovie,
    ListMovie
  },
  data() {
    return {
      movies: [],
      onMovieListPage: false
    };
  },
  computed: {
    buttonText() {
      if (this.onMovieListPage) return "Go To Movies List";
      return "Add Movie";
    },
    headertext() {
      if (this.onMovieListPage) return "Add Movie";
      return "Movies List";
    }
  },
  methods: {
    addNewMovie(movie) {
      this.movies.push(movie);
    },
    toggleScreen() {
      this.onMovieListPage = !this.onMovieListPage;
    }
  }
};
</script>

<style>
.button {
  background-color: #4caf50; /* Green */
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}
.header {
  padding: 60px;
  text-align: center;
  background: #1abc9c;
  color: white;
  font-size: 30px;
}
</style>
