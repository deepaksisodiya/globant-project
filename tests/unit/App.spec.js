import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";

describe("App.vue", () => {
  it("should be an vue instace", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.element).toMatchSnapshot();
  });
});
