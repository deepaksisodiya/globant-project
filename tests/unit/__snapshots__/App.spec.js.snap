// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should match snapshot 1`] = `
<div>
  <button>
    Add Movie
  </button>
   
  <listmovie-stub
    movies=""
  />
   
  <!---->
</div>
`;
